src/libs/actions/IOU.ts

```diff
@@ -1707,17 +1707,20 @@ function getDeleteTrackExpenseInformation(
     }
 
     if (actionableWhisperReportActionID) {
-        failureData.push({
-            onyxMethod: Onyx.METHOD.MERGE,
-            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport?.reportID}`,
-            value: {
-                [actionableWhisperReportActionID]: {
-                    originalMessage: {
-                        resolution: null,
+        const actionableWhisperReportAction = ReportActionsUtils.getReportAction(chatReportID, actionableWhisperReportActionID);
+        if (ReportActionsUtils.isActionableTrackExpense(actionableWhisperReportAction)) {
+            failureData.push({
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport?.reportID}`,
+                value: {
+                    [actionableWhisperReportActionID]: {
+                        originalMessage: {
+                            resolution: ReportActionsUtils.getOriginalMessage(actionableWhisperReportAction)?.resolution,
```

Comment:
Don't we have to pass `null` when undefined? 
```suggestion
                            resolution: ReportActionsUtils.getOriginalMessage(actionableWhisperReportAction)?.resolution ?? null,
```